### DISCUSSION TITLE
Load test

### DISCUSSION THREAD

**Discussion by NicolasHumann** [Azure/communication-ui-library#2901] [Q&A]
Hello,

We are utilizing the Web SDK to facilitate joining a Teams meeting that includes audio and video.
Our aim is to conduct a load test, which involves simulating a scenario where a large number of users (around 200) connect to the same meeting at the same time. 

Could you please share any techniques that could help us achieve this?

Thank you

**Comment by JamesBurnside** [comment]
Hi @NicolasHumann, we don't today provide a mechanism for load testing against a live Azure Communication Services service. To better understand, what is the goal of the load tests you wish to perform? What are the areas/parts of the infrastructure you looking to load test (are you looking to load test ACS specifically or parts of your own infrastructure/UI) and are you looking to run these load tests as part of a regression test suite or one-off scenarios?

**Comment by NicolasHumann** [comment]
Hello @JamesBurnside , our goal is to test our UI with many users to see the performance, regression and one-off.

**Comment by JamesBurnside** [comment]
Currently we do not provide load testing solutions for end to end scenarios. If you are using pure UI components (such as `VideoGallery`) you could pass in mocked participants alongside the actual call participants to the react component props to see what many users would look like in a live call - but this won't perform end to end scenarios where you have many live users (and all the nuances that come with many live users all having properties dynamically change).

However, we do feel this would be a great addition to have in this library but will require significant investment. To gather more needs:
* Would providing a solution where the UI is simply mocked be sufficient for your testing? Or would you be looking to have simulated live participants.
* How would you like to see interactions (e.g. mute/unmute, screenshare, turn off/on captions etc.) happen with the mocked/simulated participants? e.g. Would you want to tweak the interactions of mocked/simulated participant's yourself, have them do nothing and be static participants, or perhaps have a wide variety of complex interactions be happening automatically by the mocked/simulated participants?

**Comment by NicolasHumann** [comment]
Hello @JamesBurnside ,

Thank you for your response.
Ideally, we would like to have the ability to simulate multiple users, observe their interactions, and simulate events. This will enable us to test our UI and observe how its components respond to various events such as muting/unmuting, adding/removing participants, and so on.
